packages/xo-server-sdn-controller/src/protocol/ovsdb-client.js

```diff
@@ -295,8 +295,12 @@ export class OvsdbClient {
       'uuid-name': 'new_controller',
     })
 
-    const networks = this.host.$PIFs.map(pif => pif.$network)
+    const networks = this.host.$PIFs.map(pif => pif?.$network)
```

Comment:
There are undefined pifs in the `host.$PIFs` collection? i don't think is normal. Should we wait for their resolves?

Reply:
It can be because of a XAPI ref issue so we can't wait for it because it's possible it's NEVER resolved.

Reply:
I think we should not silently ignore valid PIFs because some NICs will not be taking into account. What do you think?

Reply:
It's a code that is frequently run, so if PIF are missing at call n, they will be resolved at call n+1.

Reply:
@benjamreis Do you know why this happens though? The host has some PIFs attached to it but the PIF objects aren't there (yet)? Isn't it an issue to just ignore them during the network creation process?

Reply:
No I don't know, I think it's a XAPI issue because after restart of XO, and disconnect/reconnect to the master the issue still happens.

Reply:
And it's not an issue to ignore them?

Reply:
It's better to ignore them than to fail because a `PIF` unrelated to the network I'm creating is a problem in XAPI.
I'll investigate this bug within XAPI and try to solve it but meanwhile I have nothing better than to work around it. :(

Reply:
The issue can come from this code https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server-sdn-controller/src/index.js#L1083-L1108.

We aren't sure that we got the PIFs records on a host update.

Reply:
I don't think so, this code is called when a new host appeared in a pool with prvate networks.
I encoutered this issue when creating a private network so all the hosts were already here.

Should I unresolved the conversation? IDK why it was solved.

Reply:
When a new host appears in a pool, its PIFs can be not fetched Yes.

The conversation was solved by me, the added comment it's just for help.

Reply:
Ok, then if it's not an issue for the network creation, I think you can just add a comment on line 300 to explain what's happening and I guess we can merge it.